frontend/src/app/pages/home/settings/settings.component.ts

```diff
@@ -117,31 +114,27 @@ export class SettingsComponent implements OnInit {
       .content(UploadAvatarComponent)
       .create();
   }
-
-  get usernameAndEmailFields() { return this.changeUsernameAndEmailForm.controls; }
+  
   get passwordFields() { return this.changePasswordForm.controls; }
   get changeProfileFields() { return this.changeProfileForm.controls; }
 
-  submitUsernameAndEmailForm() {
-    if (this.usernameAndEmailFields.email.invalid) {
-      this.alertsService.warn(`Your email must be a valid address.`);
-      return;
-    }
-
-    if (this.usernameAndEmailFields.username.invalid) {
-      this.alertsService.warn(`Your username must be between 3 and 32 characters.`);
-      return;
-    }
-    const newNameAndEmail: models.ChangeNameAndEmail = {
-      username: this.usernameAndEmailFields.username.value,
-      email: this.usernameAndEmailFields.email.value,
-      currentPassword: this.usernameAndEmailFields.currPassword.value,
+  changeEmail = (newEmail: string, password: string): Observable<string> => {
+    const changeRequest: models.ChangeEmail = {
+      currentPassword: password,
+      newEmail: newEmail
     };
-
-    this.authService.changeNameAndEmail(newNameAndEmail).subscribe(() => {
-      location.reload();
-    });
-  }
+    return this.authService.changeEmail(changeRequest)
+      .pipe(map(x => {
+        if (x) {
+          this.alertsService.success("Email successfully changed!");
+          return x.email;          
+        } else {
+          return null;
+        }
+      }), catchError(_ => {
+        return of(null);
+      }));
+  };
```

Comment:
The only "gotcha" bit about the `uploadFunction` that the `network-input` expects is that it expects a string on success, and interprets any falsy value as failure.